src/Lib/Twig/Loader.php

```diff
@@ -8,80 +7,133 @@
+ * Class Loader.
  * @package WyriHaximus\TwigView\Lib\Twig
  */
-class Loader implements \Twig_LoaderInterface
+class Loader implements LoaderInterface, SourceContextLoaderInterface
 {
-
     /**
      * Get the file contents of a template.
      *
      * @param string $name Template.
      *
      * @return string
      */
-    public function getSource($name)
+    public function getSource($name): string
     {
         $name = $this->resolveFileName($name);
+
         return file_get_contents($name);
     }
 
     /**
+     * Returns the source context for a given template logical name.
+     *
+     * @param string $name The template logical name.
+     *
+     * @throws Twig\Error\Loader When $name is not found
+     * @return Twig\Source
+     *
+     */
+    public function getSourceContext($name): Source
+    {
+        $code = $this->getSource($name);
+        $path = $this->getFilename($name);
+
+        return new Source($code, $name, $path);
+    }
+
+    /**
      * Get cache key for template.
      *
      * @param string $name Template.
      *
      * @return string
      */
-    public function getCacheKey($name)
+    public function getCacheKey($name): string
     {
         return $this->resolveFileName($name);
     }
 
     /**
      * Check if template is still fresh.
      *
-     * @param string  $name Template.
-     * @param integer $time Timestamp.
+     * @param string $name Template.
+     * @param int    $time Timestamp.
      *
-     * @return boolean
+     * @return bool
      */
-    public function isFresh($name, $time)
+    public function isFresh($name, $time): bool
     {
         $name = $this->resolveFileName($name);
+
         return filemtime($name) < $time;
     }
 
     /**
+     * Check if we have the source code of a template, given its name.
+     *
+     * @param string $name The name of the template to check if we can load.
+     *
+     * @return bool If the template source code is handled by this loader or not.
+     */
+    public function exists($name): bool
+    {
+        $filename = $this->getFilename($name);
+        if ($filename === false) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
      * Resolve template name to filename.
      *
      * @param string $name Template.
      *
+     * @throws \Twig\Error\LoaderError Thrown when template file isn't found.
      * @return string
      *
-     * @throws \Twig_Error_Loader Thrown when template file isn't found.
      */
-    // @codingStandardsIgnoreStart
-    protected function resolveFileName($name)
+    protected function resolveFileName($name): string
+    {
+        $filename = $this->getFilename($name);
+        if ($filename === false) {
+            throw new LoaderError(sprintf('Template "%s" is not defined.', $name));
+        }
+
+        return $filename;
+    }
+
+    /**
+     * Get template filename.
+     *
+     * @param string $name Template.
+     *
+     * @return string|false
```

Comment:
We can change this to string|null then in future we may be able to typehint with `: string?`